go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go

```diff
@@ -0,0 +1,456 @@
+	"vitess.io/vitess/go/vt/vterrors"
+
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/net/context"
+	"vitess.io/vitess/go/test/endtoend/cluster"
+	"vitess.io/vitess/go/vt/grpcclient"
+	"vitess.io/vitess/go/vt/log"
+	querypb "vitess.io/vitess/go/vt/proto/query"
+	vtgatepb "vitess.io/vitess/go/vt/proto/vtgate"
+	vtgateservicepb "vitess.io/vitess/go/vt/proto/vtgateservice"
+)
+
+var (
+	clusterInstance    *cluster.LocalProcessCluster
+	createVtInsertTest = `create table vt_insert_test (
+								id bigint auto_increment,
+								msg varchar(64),
+								keyspace_id bigint(20) unsigned NOT NULL,
+								primary key (id)
+								) Engine = InnoDB`
+	keyspace      = "test_keyspace"
+	hostname      = "localhost"
+	shardName     = "0"
+	cell          = "zone1"
+	certDirectory string
+	grpcCert      = ""
+	grpcKey       = ""
+	grpcCa        = ""
+	grpcName      = ""
+)
+
+func TestSecureTransport(t *testing.T) {
+	flag.Parse()
+
+	// initialize cluster
+	_, err := clusterSetUp(t)
+	assert.Nil(t, err, "setup failed")
+
+	masterTablet := *clusterInstance.Keyspaces[0].Shards[0].Vttablets[0]
+	replicaTablet := *clusterInstance.Keyspaces[0].Shards[0].Vttablets[1]
+
+	err = clusterInstance.VtctlclientProcess.InitTablet(&masterTablet, clusterInstance.Cell, keyspace, hostname, shardName)
+	assert.Nil(t, err)
+
+	// create database so vttablet can start behaving normally
+	err = masterTablet.VttabletProcess.CreateDB(keyspace)
+	assert.Nil(t, err)
+
+	err = clusterInstance.VtctlclientProcess.InitTablet(&replicaTablet, clusterInstance.Cell, keyspace, hostname, shardName)
+	assert.Nil(t, err)
+	err = replicaTablet.VttabletProcess.CreateDB(keyspace)
+	assert.Nil(t, err)
+
+	// creating table_acl_config.json file
+	tableACLConfigJSON := path.Join(certDirectory, "table_acl_config.json")
+	f, err := os.Create(tableACLConfigJSON)
+	assert.Nil(t, err)
+
+	_, err = f.WriteString(`{
+	"table_groups": [
+	{
+		"table_names_or_prefixes": ["vt_insert_test"],
+		"readers": ["vtgate client 1"],
+		"writers": ["vtgate client 1"],
+		"admins": ["vtgate client 1"]
+	}
+	]
+	}`)
+	assert.Nil(t, err)
+	err = f.Close()
+	assert.Nil(t, err)
+
+	// start the tablets
+	for _, tablet := range []cluster.Vttablet{masterTablet, replicaTablet} {
+		tablet.VttabletProcess.ExtraArgs = append(tablet.VttabletProcess.ExtraArgs, "-table-acl-config", tableACLConfigJSON, "-queryserver-config-strict-table-acl")
+		tablet.VttabletProcess.ExtraArgs = append(tablet.VttabletProcess.ExtraArgs, serverExtraArguments("vttablet-server-instance", "vttablet-client")...)
+		err = tablet.VttabletProcess.Setup()
+		assert.Nil(t, err)
+	}
+
+	// setup replication
+	var vtctlClientArgs []string
+
+	vtctlClientTmArgs := append(vtctlClientArgs, tmclientExtraArgs("vttablet-client-1")...)
+
+	// Reparenting
+	vtctlClientArgs = append(vtctlClientTmArgs, "InitShardMaster", "-force", "test_keyspace/0", masterTablet.Alias)
+	_, err = clusterInstance.VtctlProcess.ExecuteCommandWithOutput(vtctlClientArgs...)
```

Comment:
As you don't need a result here maybe write another method in VtctlProcess which just returns err?

Reply:
done